capirca/lib/paloaltofw.py

```diff
@@ -268,6 +270,21 @@ def pan_ports(ports):
         if service_name not in self.options["service"]:
           self.options["service"].append(service_name)
 
+    elif "tcp" in term.protocol or "udp" in term.protocol:
+      for p in term.protocol:
+        if p not in ["tcp", "udp"]:
+          logging.warning(
+            "WARNING: Term %s in policy %s>%s contains port-less tcp "
+            "and/or udp with protocol %s.  Move %s to another term.",
+            term.name, self.options["from_zone"][0],
+            self.options["to_zone"][0], p, p)
+          continue
+        ports = pan_ports([("0", "65535")])
+        # use prefix "" to avoid service name clash with term named "any"
+        service_name = service_map.get_service_name("any", (), ports, p, "")
```

Comment:
This doesn't handle the scenario for when only the source_port is specified, but the destination_port is not, since you're just passing () for src_ports.

Reply:
source_port only is handled in the if above: https://github.com/kevinsteves/capirca/blob/br13/capirca/lib/paloaltofw.py#L259

Reply:
Great just saw that this term is correct.

```
<entry name="service-accept-traceroute-source-ports-udp">
              <protocol>
                <udp>
                  <port>0-65535</port>
                  <source-port>33434-33534</source-port>
                </udp>
              </protocol>
            </entry>
```